### DISCUSSION TITLE
prior working app stopped working with assertionerrors

### DISCUSSION THREAD

**Discussion by gjelsas** [flet-dev/flet#3334] [Q&A] [answered]
My prior running Program stopped working with Errors I cannot understand. Starting the program leads to errors below, and no screen is shown at all.

I start the program the way I always did: 
- source the venv and 
- run the app. 

Is this a Flutter thing? 
I'm running Nixos by the way, could it be a problem of the OS ?

I'd appreciate any hint!

this is the rather lengthy output:

> ** (flet:14146): CRITICAL **: 10:36:28.629: FlBinaryMessenger *fl_engine_get_binary_messenger(FlEngine *): assertion 'FL_IS_ENGINE(self)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlTextureRegistrar *fl_engine_get_texture_registrar(FlEngine *): assertion 'FL_IS_ENGINE(self)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlPluginRegistrar *fl_plugin_registrar_new(FlView *, FlBinaryMessenger *, FlTextureRegistrar *): assertion 'FL_IS_BINARY_MESSENGER(messenger)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlBinaryMessenger *fl_plugin_registrar_get_messenger(FlPluginRegistrar *): assertion 'FL_IS_PLUGIN_REGISTRAR(self)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlMethodChannel *fl_method_channel_new(FlBinaryMessenger *, const gchar *, FlMethodCodec *): assertion 'FL_IS_BINARY_MESSENGER(messenger)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlMethodChannel *fl_method_channel_new(FlBinaryMessenger *, const gchar *, FlMethodCodec *): assertion 'FL_IS_BINARY_MESSENGER(messenger)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlEventChannel *fl_event_channel_new(FlBinaryMessenger *, const gchar *, FlMethodCodec *): assertion 'FL_IS_BINARY_MESSENGER(messenger)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: void fl_method_channel_set_method_call_handler(FlMethodChannel *, FlMethodChannelMethodCallHandler, gpointer, GDestroyNotify): assertion 'FL_IS_METHOD_CHANNEL(self)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: void fl_method_channel_set_method_call_handler(FlMethodChannel *, FlMethodChannelMethodCallHandler, gpointer, GDestroyNotify): assertion 'FL_IS_METHOD_CHANNEL(self)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlBinaryMessenger *fl_engine_get_binary_messenger(FlEngine *): assertion 'FL_IS_ENGINE(self)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlTextureRegistrar *fl_engine_get_texture_registrar(FlEngine *): assertion 'FL_IS_ENGINE(self)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlPluginRegistrar *fl_plugin_registrar_new(FlView *, FlBinaryMessenger *, FlTextureRegistrar *): assertion 'FL_IS_BINARY_MESSENGER(messenger)' failed
> package:media_kit_libs_linux registered.
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlBinaryMessenger *fl_engine_get_binary_messenger(FlEngine *): assertion 'FL_IS_ENGINE(self)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlTextureRegistrar *fl_engine_get_texture_registrar(FlEngine *): assertion 'FL_IS_ENGINE(self)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlPluginRegistrar *fl_plugin_registrar_new(FlView *, FlBinaryMessenger *, FlTextureRegistrar *): assertion 'FL_IS_BINARY_MESSENGER(messenger)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlBinaryMessenger *fl_plugin_registrar_get_messenger(FlPluginRegistrar *): assertion 'FL_IS_PLUGIN_REGISTRAR(self)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlMethodChannel *fl_method_channel_new(FlBinaryMessenger *, const gchar *, FlMethodCodec *): assertion 'FL_IS_BINARY_MESSENGER(messenger)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: void fl_method_channel_set_method_call_handler(FlMethodChannel *, FlMethodChannelMethodCallHandler, gpointer, GDestroyNotify): assertion 'FL_IS_METHOD_CHANNEL(self)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlTextureRegistrar *fl_plugin_registrar_get_texture_registrar(FlPluginRegistrar *): assertion 'FL_IS_PLUGIN_REGISTRAR(self)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlView *fl_plugin_registrar_get_view(FlPluginRegistrar *): assertion 'FL_IS_PLUGIN_REGISTRAR(self)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlBinaryMessenger *fl_engine_get_binary_messenger(FlEngine *): assertion 'FL_IS_ENGINE(self)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlTextureRegistrar *fl_engine_get_texture_registrar(FlEngine *): assertion 'FL_IS_ENGINE(self)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlPluginRegistrar *fl_plugin_registrar_new(FlView *, FlBinaryMessenger *, FlTextureRegistrar *): assertion 'FL_IS_BINARY_MESSENGER(messenger)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlBinaryMessenger *fl_engine_get_binary_messenger(FlEngine *): assertion 'FL_IS_ENGINE(self)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlTextureRegistrar *fl_engine_get_texture_registrar(FlEngine *): assertion 'FL_IS_ENGINE(self)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlPluginRegistrar *fl_plugin_registrar_new(FlView *, FlBinaryMessenger *, FlTextureRegistrar *): assertion 'FL_IS_BINARY_MESSENGER(messenger)' failed
> 
> (flet:14146): GLib-GObject-CRITICAL **: 10:36:28.629: g_object_ref: assertion 'G_IS_OBJECT (object)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlBinaryMessenger *fl_plugin_registrar_get_messenger(FlPluginRegistrar *): assertion 'FL_IS_PLUGIN_REGISTRAR(self)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlMethodChannel *fl_method_channel_new(FlBinaryMessenger *, const gchar *, FlMethodCodec *): assertion 'FL_IS_BINARY_MESSENGER(messenger)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: void fl_method_channel_set_method_call_handler(FlMethodChannel *, FlMethodChannelMethodCallHandler, gpointer, GDestroyNotify): assertion 'FL_IS_METHOD_CHANNEL(self)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlBinaryMessenger *fl_engine_get_binary_messenger(FlEngine *): assertion 'FL_IS_ENGINE(self)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlTextureRegistrar *fl_engine_get_texture_registrar(FlEngine *): assertion 'FL_IS_ENGINE(self)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlPluginRegistrar *fl_plugin_registrar_new(FlView *, FlBinaryMessenger *, FlTextureRegistrar *): assertion 'FL_IS_BINARY_MESSENGER(messenger)' failed
> 
> (flet:14146): GLib-GObject-CRITICAL **: 10:36:28.629: g_object_ref: assertion 'G_IS_OBJECT (object)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlBinaryMessenger *fl_plugin_registrar_get_messenger(FlPluginRegistrar *): assertion 'FL_IS_PLUGIN_REGISTRAR(self)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlMethodChannel *fl_method_channel_new(FlBinaryMessenger *, const gchar *, FlMethodCodec *): assertion 'FL_IS_BINARY_MESSENGER(messenger)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: void fl_method_channel_set_method_call_handler(FlMethodChannel *, FlMethodChannelMethodCallHandler, gpointer, GDestroyNotify): assertion 'FL_IS_METHOD_CHANNEL(self)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlBinaryMessenger *fl_engine_get_binary_messenger(FlEngine *): assertion 'FL_IS_ENGINE(self)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlTextureRegistrar *fl_engine_get_texture_registrar(FlEngine *): assertion 'FL_IS_ENGINE(self)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlPluginRegistrar *fl_plugin_registrar_new(FlView *, FlBinaryMessenger *, FlTextureRegistrar *): assertion 'FL_IS_BINARY_MESSENGER(messenger)' failed
> 
> (flet:14146): GLib-GObject-CRITICAL **: 10:36:28.629: g_object_ref: assertion 'G_IS_OBJECT (object)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlView *fl_plugin_registrar_get_view(FlPluginRegistrar *): assertion 'FL_IS_PLUGIN_REGISTRAR(self)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlView *fl_plugin_registrar_get_view(FlPluginRegistrar *): assertion 'FL_IS_PLUGIN_REGISTRAR(self)' failed
> 
> (flet:14146): GLib-GObject-CRITICAL **: 10:36:28.629: invalid (NULL) pointer instance
> 
> (flet:14146): GLib-GObject-CRITICAL **: 10:36:28.629: g_signal_handler_find: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlView *fl_plugin_registrar_get_view(FlPluginRegistrar *): assertion 'FL_IS_PLUGIN_REGISTRAR(self)' failed
> 
> (flet:14146): GLib-GObject-CRITICAL **: 10:36:28.629: invalid (NULL) pointer instance
> 
> (flet:14146): GLib-GObject-CRITICAL **: 10:36:28.629: g_signal_connect_data: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlView *fl_plugin_registrar_get_view(FlPluginRegistrar *): assertion 'FL_IS_PLUGIN_REGISTRAR(self)' failed
> 
> (flet:14146): GLib-GObject-CRITICAL **: 10:36:28.629: invalid (NULL) pointer instance
> 
> (flet:14146): GLib-GObject-CRITICAL **: 10:36:28.629: g_signal_connect_data: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlView *fl_plugin_registrar_get_view(FlPluginRegistrar *): assertion 'FL_IS_PLUGIN_REGISTRAR(self)' failed
> 
> (flet:14146): GLib-GObject-CRITICAL **: 10:36:28.629: invalid (NULL) pointer instance
> 
> (flet:14146): GLib-GObject-CRITICAL **: 10:36:28.629: g_signal_connect_data: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlView *fl_plugin_registrar_get_view(FlPluginRegistrar *): assertion 'FL_IS_PLUGIN_REGISTRAR(self)' failed
> 
> (flet:14146): GLib-GObject-CRITICAL **: 10:36:28.629: invalid (NULL) pointer instance
> 
> (flet:14146): GLib-GObject-CRITICAL **: 10:36:28.629: g_signal_connect_data: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlView *fl_plugin_registrar_get_view(FlPluginRegistrar *): assertion 'FL_IS_PLUGIN_REGISTRAR(self)' failed
> 
> (flet:14146): GLib-GObject-CRITICAL **: 10:36:28.629: invalid (NULL) pointer instance
> 
> (flet:14146): GLib-GObject-CRITICAL **: 10:36:28.629: g_signal_connect_data: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlView *fl_plugin_registrar_get_view(FlPluginRegistrar *): assertion 'FL_IS_PLUGIN_REGISTRAR(self)' failed
> 
> (flet:14146): GLib-GObject-CRITICAL **: 10:36:28.629: invalid (NULL) pointer instance
> 
> (flet:14146): GLib-GObject-CRITICAL **: 10:36:28.629: g_signal_connect_data: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlView *fl_plugin_registrar_get_view(FlPluginRegistrar *): assertion 'FL_IS_PLUGIN_REGISTRAR(self)' failed
> 
> (flet:14146): GLib-GObject-CRITICAL **: 10:36:28.629: invalid (NULL) pointer instance
> 
> (flet:14146): GLib-GObject-CRITICAL **: 10:36:28.629: g_signal_connect_data: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlView *fl_plugin_registrar_get_view(FlPluginRegistrar *): assertion 'FL_IS_PLUGIN_REGISTRAR(self)' failed
> 
> (flet:14146): GLib-GObject-CRITICAL **: 10:36:28.629: invalid (NULL) pointer instance
> 
> (flet:14146): GLib-GObject-CRITICAL **: 10:36:28.629: g_signal_connect_data: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlView *fl_plugin_registrar_get_view(FlPluginRegistrar *): assertion 'FL_IS_PLUGIN_REGISTRAR(self)' failed
> 
> (flet:14146): GLib-GObject-CRITICAL **: 10:36:28.629: invalid (NULL) pointer instance
> 
> (flet:14146): GLib-GObject-CRITICAL **: 10:36:28.629: g_signal_connect_data: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlView *fl_plugin_registrar_get_view(FlPluginRegistrar *): assertion 'FL_IS_PLUGIN_REGISTRAR(self)' failed
> 
> (flet:14146): Gtk-CRITICAL **: 10:36:28.629: gtk_container_forall: assertion 'GTK_IS_CONTAINER (container)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlBinaryMessenger *fl_plugin_registrar_get_messenger(FlPluginRegistrar *): assertion 'FL_IS_PLUGIN_REGISTRAR(self)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlMethodChannel *fl_method_channel_new(FlBinaryMessenger *, const gchar *, FlMethodCodec *): assertion 'FL_IS_BINARY_MESSENGER(messenger)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: void fl_method_channel_set_method_call_handler(FlMethodChannel *, FlMethodChannelMethodCallHandler, gpointer, GDestroyNotify): assertion 'FL_IS_METHOD_CHANNEL(self)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlBinaryMessenger *fl_engine_get_binary_messenger(FlEngine *): assertion 'FL_IS_ENGINE(self)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlTextureRegistrar *fl_engine_get_texture_registrar(FlEngine *): assertion 'FL_IS_ENGINE(self)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlPluginRegistrar *fl_plugin_registrar_new(FlView *, FlBinaryMessenger *, FlTextureRegistrar *): assertion 'FL_IS_BINARY_MESSENGER(messenger)' failed
> 
> (flet:14146): GLib-GObject-CRITICAL **: 10:36:28.629: g_object_ref: assertion 'G_IS_OBJECT (object)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlBinaryMessenger *fl_plugin_registrar_get_messenger(FlPluginRegistrar *): assertion 'FL_IS_PLUGIN_REGISTRAR(self)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: FlMethodChannel *fl_method_channel_new(FlBinaryMessenger *, const gchar *, FlMethodCodec *): assertion 'FL_IS_BINARY_MESSENGER(messenger)' failed
> 
> ** (flet:14146): CRITICAL **: 10:36:28.629: void fl_method_channel_set_method_call_handler(FlMethodChannel *, FlMethodChannelMethodCallHandler, gpointer, GDestroyNotify): assertion 'FL_IS_METHOD_CHANNEL(self)' failed
> 
> 

Thanks for the wonderful flet package and the work you are doing!

**Comment by FeodorFitsner** [comment]
Thanks!

Since it's NixOS, and issue filed without a template, and I'm moving it to discussions.

Is Flutter supported on NixOS? What version of Flutter SDK do you have? Have you tried running `flet build linux` to see if it's buildable?

**Reply by gjelsas** [reply]
`flet build linux`  gives me 

> Creating Flutter bootstrap project from gh:flet-dev/flet-build-template with ref 0.2.0...git clone failed with error: Fehler: Pfadspezifikation '0.2.0' stimmt mit keinen Git bekannten Dateien überein
> 
> Command '['git', 'checkout', '0.2.0']' returned non-zero exit status 1.

The german 

> Fehler: Pfadspezifikation '0.2.0' stimmt mit keinen Git bekannten Dateien überein 

means that git can, for the path 0.2.0, not find matching files.

**Comment by gjelsas** [comment]
Thank you for the fast Response!

I did a new setup in an ubuntu VM and it works as it was supposed to work. So it's definitely a Nixos problem. Although it also did work on Nixos before... 

I'm investigating a Nixos solution for this from here on. Maybe this comment will help others not blaming flet for this kind of error.

Thanks again!

**Comment by mse11** [comment]
Hi I face same issue. Usually I do 'flutter upgrade' so in my case:

PROBLEM:

$ flutter upgrade
Upgrading Flutter to 3.22.1 from 3.19.5 in /home/msestrie/MSE/MSEsw/flutter...
Downloading Linux x64 Dart SDK from Flutter engine 55eae6864b296dd9f43b2cc7577ec256e5c32a8d...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0  221M    0  5551    0     0  26687      0  2:24:55 --:--:--  2:24:55 26559
 11  221M   11 25.8M    0     0  23.1M      0  0:00:09  0:00:01  0:00:08 23.1M
 29  221M   29 64.2M    0     0  30.4M      0  0:00:07  0:00:02  0:00:05 30.4M
 46  221M   46  103M    0     0  33.3M      0  0:00:06  0:00:03  0:00:03 33.2M
 64  221M   64  142M    0     0  34.6M      0  0:00:06  0:00:04  0:00:02 34.6M
 82  221M   82  181M    0     0  35.5M      0  0:00:06  0:00:05  0:00:01 37.0M
 99  221M   99  220M    0     0  36.0M      0  0:00:06  0:00:06 --:--:-- 38.8M
100  221M  100  221M    0     0  36.0M      0  0:00:06  0:00:06 --:--:-- 39.0M
Building flutter tool...
Resolving dependencies...
Downloading packages...
Got dependencies.
 
HOW I FIX IT:
$ flutter downgrade 3.19.5
$ flutter clean
$ flutter build linux

And works

**Reply by gjelsas** [reply]
Thanks for your answer! 

I cannot up- or downgrade with the flutter tools, because flutter is on a readonly filesystem on Nixos. Its installed in a nix-shell environment. I tried Flutter versions 3.16 3.19 and 3.22 but without success... Are you developing with flet on Nixos? If so what are you using flakes, or nix-shell or something else?

**Reply by mse11** [reply]
Ubuntu 20.04 VM Guest

**Comment by mse11** [comment]
@gjelsas Because I found this [flutter issue](https://github.com/flutter/flutter/issues/144873) so I tried to investigate:

I upgrade again my flutter version to previously **problematic one 3.22.1** - see error like :
 ```
FlBinaryMessenger *fl_engine_get_binary_messenger(FlEngine *): assertion 'FL_IS_ENGINE(self)'):
   bla bla ...

```
So state on my machine:
```
msestrie@MSE ~/MSE/MSEprojects/PYTHON/flet_ECO/flet/client(main) $ flutter --version
Flutter 3.22.1 • channel stable • https://github.com/flutter/flutter.git
Framework • revision a14f74ff3a (2 days ago) • 2024-05-22 11:08:21 -0500
Engine • revision 55eae6864b
Tools • Dart 3.4.1 • DevTools 2.34.3
msestrie@MSE ~/MSE/MSEprojects/PYTHON/flet_ECO/flet/client(main) $

```
Then check 3.22.1 release hello world [example file](https://github.com/flutter/flutter/blob/3.22.1/examples/hello_world/linux/my_application.cc)  with assumption it should work somehow and just tried to use it in flet. Just copy/paste patch :) and app works :). For better understanding see picture :

![flet_CONTRIBUTION_flutter_issue](https://github.com/flet-dev/flet/assets/24617949/419b44a4-6c54-4f75-ac3f-9f66044d312b)

Hope helps

NOTE: Sometimes I have issue with first app run, but re-launch works (but I notice first run issue also from previous versions, so it is not related this "patch" ... ).

**Comment by gjelsas** [comment]
It seems it is a [flutter issue](https://github.com/ubuntu/yaru.dart/issues/903) and they are already working on the fix. So I'm still stuck on my Ubuntu-VM as I'm not able to get flutter 3.19 to work on my Nixos machine...

**Comment by gjelsas** [comment, marked as answer]
Okay, I found a way using the nixos overlay syntax to create a flutter 3.19 overlay and start my program. For anyone who is interested here are my shell.nix and the flutter-overlay.nix files. They are in the same folder. You can remove the shell-hooks, if you don't want to start the Program or source the virtual environment on activation of the nix-shell environment.

to start execute `nix-shell` in the folder containing the two files.

shell.nix:

```
{ pkgs ? import <nixpkgs> { 
overlays = [ (import ./flutter-overlay.nix )]; 
} }:
pkgs.mkShellNoCC {
  packages = with pkgs; [
	python311Packages.flet
  ];
  shellHook = ''
  	source /home/user/path/to/your/venv/bin/activate
  	python /home/user/path/to/your/flet/program.py
  '';
}

```

 
flutter-overlay.nix:
`
self: super:
{
  flutter = super.flutter319;
}
`
